### PR TITLE
feat: trim string values and interpret empty string value as null M8771

### PR DESCRIPTION
**Pull request by mswertz**
How to test
* will trim spaces in csv and excel (except for columnType=text)
* will trim spaces in when you edit data via web forms (except for columnType=text)

N.B. is implemented server side so is trimmed on every insert, update

### CODE REVIEW

File: backend/molgenis-emx2/src/main/java/org/molgenis/emx2/ColumnType.java [hunk 1]
```diff
@@ -120,4 +120,8 @@ public boolean isFile() {
   public boolean isHeading() {
     return HEADING.equals(getBaseType());
   }
+
+  public boolean isAtomicType() {
```
Comment (connoratrug) [inline backend/molgenis-emx2/src/main/java/org/molgenis/emx2/ColumnType.java:124]: why not use a class AtomicType instead of a function

**Review by erikzwart — APPROVED** [review]
In pet store added a descrription 'TEXT' to test behaviour:
[pet store_backup.zip](https://github.com/molgenis/molgenis-emx2/files/8015990/pet.store_backup.zip)

Mock order with pre and postfix whitespaces:
[Order.zip](https://github.com/molgenis/molgenis-emx2/files/8015998/Order.zip)

Whitespaces are removed from columnType INT, DECIMAL, STRING but not TEXT as intended
[Order_after_update.csv](https://github.com/molgenis/molgenis-emx2/files/8016014/Order_after_update.csv)

Insert and update from web ui also removes whitespaces as expected:
image

File: backend/molgenis-emx2/src/main/java/org/molgenis/emx2/ColumnType.java (hunk 1 above)
Comment (connoratrug) [reply backend/molgenis-emx2/src/main/java/org/molgenis/emx2/ColumnType.java:124]: a never mind then , although this is starting to be a pretty complex enum ( with types and subtypes)

Comment (connoratrug) [reply backend/molgenis-emx2/src/main/java/org/molgenis/emx2/ColumnType.java:124]: could still use a interface , but then i looked at the usage and see this is only used in a test context, would consider removing the type test

**Comment by erikzwart** [comment]
[#8771 As datamanager I would like to have spaces removed at the beginning/end of all values imported from CSV or Excel](https://trello.com/c/Eo7BkMIh/398-8771-as-datamanager-i-would-like-to-have-spaces-removed-at-the-beginning-end-of-all-values-imported-from-csv-or-excel)

**Pull request merged**